pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -38,12 +39,33 @@ func NewAnnotationHistorian(annotations annotations.Repository, dashboards dashb
 func (h *AnnotationStateHistorian) RecordStatesAsync(ctx context.Context, rule *ngmodels.AlertRule, states []state.StateTransition) {
 	logger := h.log.FromContext(ctx)
 	// Build annotations before starting goroutine, to make sure all data is copied and won't mutate underneath us.
-	annotations := h.buildAnnotations(rule, states, logger)
+	annotations := buildAnnotations(rule, states, logger)
 	panel := parsePanelKey(rule, logger)
 	go h.recordAnnotationsSync(ctx, panel, annotations, logger)
 }
 
-func (h *AnnotationStateHistorian) buildAnnotations(rule *ngmodels.AlertRule, states []state.StateTransition, logger log.Logger) []annotations.Item {
+func (h *AnnotationStateHistorian) recordAnnotationsSync(ctx context.Context, panel *panelKey, annotations []annotations.Item, logger log.Logger) {
+	if panel != nil {
+		dashID, err := h.dashboards.getID(ctx, panel.orgID, panel.dashUID)
+		if err != nil {
+			logger.Error("Error getting dashboard for alert annotation", "dashboardUID", panel.dashUID, "error", err)
+			return
+		}
+
+		for i := range annotations {
+			annotations[i].DashboardId = dashID
+			annotations[i].PanelId = panel.panelID
+		}
+	}
+
+	if err := h.annotations.SaveMany(ctx, annotations); err != nil {
+		logger.Error("Error saving alert annotation batch", "error", err)
```

Comment:
same with these, do we want to fail silent from the perspective of the caller? How do we also test this?

Reply:
Yes, that's the intended behavior.

Also, I think git got a little confused on the diff here. This code was simply moved to be higher in the file, not changed. It's nothing new, you can see it exactly matches the code that got removed from line 97 in the old version.

Reply:
I wonder if it should be followed by `return` so we do not log debug in error case

Reply:
Good idea, I used a slightly different pattern in the Loki implementation that removes ambiguity and prevents this whole class of issue.

Reply:
This was done implicitly in a different PR, this diff is no longer part of this PR.